lib/search-bar/index.tsx

```diff
@@ -14,33 +14,42 @@ import NewNoteIcon from '../icons/new-note';
 import SearchField from '../search-field';
 import MenuIcon from '../icons/menu';
 import { withoutTags } from '../utils/filter-notes';
-import { createNote, search } from '../state/ui/actions';
+import { createNote, search, toggleNavigation } from '../state/ui/actions';
 
 import * as S from '../state';
+import * as T from '../types';
 
-const { newNote, toggleNavigation } = appState.actionCreators;
+const { newNote } = appState.actionCreators;
 const { recordEvent } = tracks;
 
 type OwnProps = {
   onNewNote: Function;
-  onToggleNavigation: Function;
+  noteBucket: object;
+  onNoteOpened: Function;
 };
 
 type StateProps = {
   searchQuery: string;
   showTrash: boolean;
 };
 
-type Props = OwnProps & StateProps;
+type DispatchProps = {
+  toggleNavigation: () => any;
+};
+
+type Props = OwnProps & StateProps & DispatchProps;
 
-export const SearchBar: FunctionComponent<Props> = ({
+export const SearchBar: Component<Props> = ({
   onNewNote,
-  onToggleNavigation,
   searchQuery,
   showTrash,
 }) => (
   <div className="search-bar theme-color-border">
-    <IconButton icon={<MenuIcon />} onClick={onToggleNavigation} title="Menu" />
+    <IconButton
+      icon={<MenuIcon />}
+      onClick={() => toggleNavigation()}
```

Comment:
should be safe to pass just the function itself instead of creating a new one. I'm pretty sure you'd catch this but I wanted to point it out since I saw it.

```ts
onClick={toggleNavigation}
```

Reply:
This was an artifact of me not being able to get it to call the dang thing. I just figured out that I need to pass it in above.. but I don't understand why...